### PR TITLE
Service/time: implement posix time to calendar conversion

### PR DESCRIPTION
**Pull request by mailwl**
addittionally, some changes in service/time structures

### CODE REVIEW

File: src/core/hle/kernel/hle_ipc.cpp [hunk 1]
```diff
@@ -350,7 +350,7 @@ std::string HLERequestContext::Description() const {
         }
         s << ']';
     }
-    s << ", data_size:" << command_header->data_size.Value();
+    s << ", data_size: 0x" << command_header->data_size.Value();
```
Comment (Hexagon12) [inline src/core/hle/kernel/hle_ipc.cpp]: Unintentional change?

Comment (mailwl) [reply src/core/hle/kernel/hle_ipc.cpp]: yes, data_size is printed as hex, and may confuse. will remove

File: src/core/hle/service/time/time.cpp [hunk 2]
```diff
@@ -103,23 +106,69 @@ class ITimeZoneService final : public ServiceFramework<ITimeZoneService> {
...
         rb.Push(RESULT_SUCCESS);
         rb.PushRaw(calendar_time);
         rb.PushRaw(additional_info);
     }
+
+    void PosixToCalendar(u64 posix_time, CalendarTime& calendar_time,
+                         CalendarAdditionalInfo& additional_info, const TimeZoneRule& /*rule*/) {
+        std::time_t t(posix_time);
+        std::tm* tm = std::localtime(&t);
+        if (tm) {
```
Comment (lioncash) [inline src/core/hle/service/time/time.cpp]:
```cpp
if (!tm) {
    return;
}
```

**Comment by mailwl** [comment]
fixed

**Pull request merged**